src/analyze/BuildSystem.re

```diff
@@ -124,27 +124,52 @@ let detect = (rootPath, bsconfig) => {
   }) : Bsb(bsbVersion);
 };
 
-let getEsyCompiledBase = esyVersion =>
+let getEsyCompiledBase = (root, esyVersion) =>
   switch (String.split_on_char('.', esyVersion)) {
   | [_, minor, ..._] =>
-    if(int_of_string(minor) >= 3) {
-      "_esy" /+ "default" /+ "build"
-    } else {
-      "_build" /* TODO */
+    let env = Unix.environment () |. Array.to_list;
+    try {
+      let root = List.find(var => Utils.startsWith(var, "cur__original_root="), env);
```

Comment:
maybe ditch this part and go right to the `execResult`? This function should only be run on startup

Reply:
We wanna check the environment to support the aliased sandboxes use case. Why do you suggest skipping this check?

Reply:
I guess I'm confused about why vscode's environment would have these variables. Do you expect people to start vscode with `esy code .` or similar?

Reply:
huhh ok!